### PR TITLE
Feat: add helm repo list endpoint

### PR DESCRIPTION
**Pull request by wangyikewxgm**
add helm repo list interface
Signed-off-by: 楚岳 <wangyike.wyk@alibaba-inc.com>

### Description of your changes

Fixes #

I have:

- [ ] Read and followed KubeVela's [contribution process](https://github.com/oam-dev/kubevela/blob/master/contribute/create-pull-request.md).
- [ ] [Related Docs](https://github.com/oam-dev/kubevela.io) updated properly. In a new feature or configuration option, an update to the documentation is necessary. 
- [ ] Run `make reviewable` to ensure this PR is ready for review.
- [ ] Added `backport release-x.y` labels to auto-backport this PR if necessary.

### CODE REVIEW

**Review by zzxwill — APPROVED** [review]

File: pkg/apiserver/rest/apis/v1/types.go [hunk 1]
```diff
@@ -1277,3 +1277,14 @@ type LoginUserInfoResponse struct {
...
+
+// ChartRepoResponse the response body of  chart repo
+type ChartRepoResponse struct {
+	URL        string `json:"url"`
+	SecretName string `json:"secretName"`
+}
+
+// ChartRepoResponseList the response body of list chart repo
+type ChartRepoResponseList struct {
+	ChartRepoResponse []*ChartRepoResponse `json:"chartRepoResponse"`
```
Comment (wonderflow) [inline pkg/apiserver/rest/apis/v1/types.go:1289]:
```suggestion
	ChartRepoResponse []*ChartRepoResponse `json:"repos"`
```

**Review by wonderflow — APPROVED** [review]
LGTM after nit

File: pkg/apiserver/rest/webservice/helm.go [hunk 2]
```diff
@@ -46,6 +46,15 @@ func (h helmWebService) GetWebService() *restful.WebService {
 
 	tags := []string{"repository", "helm"}
 
+	// List charts
+	ws.Route(ws.GET("").To(h.listRepo).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/helm.go:50]:
```suggestion
	ws.Route(ws.GET("/chart_repos").To(h.listChartRepo).
```

File: pkg/apiserver/rest/webservice/helm.go [hunk 3]
```diff
@@ -46,6 +46,15 @@ func (h helmWebService) GetWebService() *restful.WebService {
...
+		Doc("list chart repo").
+		Metadata(restfulspec.KeyOpenAPITags, tags).
+		Param(ws.QueryParameter("project", "the config project").DataType("string")).
+		Returns(200, "OK", []string{}).
+		Returns(400, "Bad Request", bcode.Bcode{}).
+		Writes([]string{}))
+
 	// List charts
 	ws.Route(ws.GET("/charts").To(h.listCharts).
```
Comment (barnettZQG) [inline pkg/apiserver/rest/webservice/helm.go:59]: support get charts by repo URL and secret name

File: pkg/apiserver/rest/apis/v1/types.go [hunk 4]
```diff
@@ -1277,3 +1277,14 @@ type LoginUserInfoResponse struct {
 	PlatformPermissions []PermissionBase            `json:"platformPermissions"`
 	ProjectPermissions  map[string][]PermissionBase `json:"projectPermissions"`
 }
+
+// ChartRepoResponse the response body of  chart repo
+type ChartRepoResponse struct {
```
Comment (barnettZQG) [inline pkg/apiserver/rest/apis/v1/types.go:1282]: missing RepoType?

Comment (wangyikewxgm) [reply pkg/apiserver/rest/apis/v1/types.go:1282]: why need this?

**Review by barnettZQG — APPROVED** [review]
LGTM

**Pull request merged**